src/ixdat/techniques/deconvolution.py

```diff
@@ -1,217 +1,150 @@
+            Henry volatility constant in water in Molecule data from siq.-CURRENTLY WRONG
+            chip: Optional. Needed to define capillary flow. Default will use
+                SpectroInlets chip from siq.
+            carrier_gas: The carrier gas used to calculate capillary flow. Defaults to He
+            gas_volume: the volume of the headspace volume in the chip. Default is the
+                volume of the SpectroInlets chip.
         """
-
-        if MS_data and parameters:  # TODO: Make two different classes
-            raise Exception(
-                "Kernel can only be initialized with data OR parameters, not both"
-            )
-        if EC_data and MS_data:
-            print("Generating kernel from measured data")
+        if data is not None:
+            self.mol = mol
+            self.data = data
             self.type = "measured"
-        elif parameters:
-            print("Generating kernel from parameters")
+            print("Generating `ECMSImpulseResponse` from measured data.")
+            if working_distance is not None or A_el is not None:
+                raise UserWarning(
+                    "Data was used to generate `ECMSImpulseResponse` ignoring the given"
+                    "working_distance/electrode area."
+                )
+
+        elif working_distance is not None and A_el is not None:
+            self.mol = mol
             self.type = "functional"
+            if not plugins.use_siq:
+                raise TechniqueError(
+                    "`ECMSImpulseResponse` will only work properly when using "
+                    # TODO this should be improved.- doesnt need to fully depend on siq
+                    # integration
+                    "`spectro_inlets_quantification` "
+                    "(`ixdat.plugins.activate_siq()`). "
+                )
+            # calculate the capillary flow for the specified gas & chip
+            Chip = plugins.siq.Chip
+            chip = chip or Chip()
+            n_dot = chip.calc_n_dot_0(gas=carrier_gas)
+            # convert to volumetric flux using T and p given by chip.
+            V_dot = n_dot * R * chip.T / chip.p
+            # find the other parameters from the siq Molecule files
+            Molecule = plugins.siq.Molecule
+            molecule = Molecule(mol)
+            # print(molecule)
+            if diff_const is None:
+                raise TechniqueError("Default diffusion constant not yet implemented")
+                diff_const = molecule.D
+                # TODO double check units and understand why siq integration is not
+                # working
+                print(diff_const)
+            if henry_vola is None:
+                raise TechniqueError("Default henry volatility not yet implemented")
+                # henry_vola = molecule.H_0
+                # TODO convert this to the right unit! (dimensionless henry volatility)
```

Comment:
Sanders doesnt recommend H_0 anymore (?)
I think this actually sounds pretty sensible: https://www.degruyter.com/document/doi/10.1515/pac-2020-0302/html
And is the IUPAC recommendation - so why not use that?
I'll rename `henry_vola` to `H_v_cc`

Reply:
similarly, H_0 should then be called H_s_cp with the only problem that this will be in [mol/(m^3*Pa)], which is not the same unit as what NIST gives. Not sure what to do about that. But I do like the naming system Sanders has on his website now